### ISSUE TITLE
Folds don't open when moving to a mark inside them for the second time

### ISSUE THREAD

**Issue by muniter** [neovim/neovim#15873]
### Neovim version (nvim -v)

0.6.0-dev+358-g6736ee8be

### Vim (not Nvim) behaves the same?

yes

### Operating system/version

Linux, Arch

### Terminal name/version

Alacritty

### $TERM environment variable

xterm-256color

### Installation

build from repo

### How to reproduce the issue

nvim -u NONE
v0.6.0-dev+358-g6736ee8be
Create the following file
```
a
b <- Put the cursor here
c <-Mark this positon mc
```
With the cursor at "b" do "zfG" which folds to the end, move to the mark "c" positioned at c, it moves the cursor and opens the fold, close the fold again "zc", we are back at the position of "b", move to the mark "c" again and it does nothing, while it should move the cursor again and open the fold.

**TLDR**: `"ia^Mb^Mc^]mckzfG'azc'a`

### Expected behavior

Fold should open every time that the cursor is moved.

### Actual behavior

Opens fold the first time, but if the fold gets closed using "zc" and you try to go again it fails opening the fold.

**Comment by justinmk** [comment]
related? #15490

**Comment by muniter** [comment]
> related? #15490 

Don't think so, the issue is with the logic in normal.c `nv_gomark` at the end decides whether to open the fold, seems like the logic is bogus. 

I suspect of this line
https://github.com/neovim/neovim/blob/79fb9ed080bb32a442d1c788da6f0b71d72bcedd/src/nvim/normal.c#L6627

I actually ran into this while writing a "mark_spec" test file for the expected current behavior of marks, to make sure #15831 won't break expected behavior.

**Comment by muniter** [comment]
After more consideration I don't think this is a bug, is just the way folding works.

When using `zc` the fold closes but the cursor keeps it's position. In this case the first movement to the mark opens the fold because the cursor moved. The next attempts to move to the mark do not move the cursor therefore the fold doesn't open.

I was under the impression that closing a fold moved the cursor to the first line which is not the case.
